src/posthog-featureflags.ts

```diff
@@ -199,15 +199,20 @@ export class PostHogFeatureFlags {
             callback: (response) => {
                 this.setReloadingPaused(false)
 
-                if (response.statusCode !== 200) {
-                    return // or error out??
-                }
-                // reset anon_distinct_id after at least a single request with it
-                // makes it through
-                this.$anon_distinct_id = undefined
-                if (response.json) {
-                    this.receivedFeatureFlags(response.json)
+                let errorsLoading = true
+
+                if (response.statusCode === 200) {
+                    // successful request
+                    // reset anon_distinct_id after at least a single request with it
+                    // makes it through
+                    this.$anon_distinct_id = undefined
+                    errorsLoading = false
                 }
+                // :TRICKY: We want to fire the callback even if the request fails
```

Comment:
This feels like a hack... Why call `receivedFeatureFlags` when from what I can see the only thing it needs is `this._fireFeatureFlagsCallbacks(errorsLoading)` - why not just call that directly?

Reply:
I also want all the other stuff that this function does (like setting decideEndpointWasHit) which then warns users when they call `getFeatureFlag()` that flags didn't load in time.